solr/core/src/java/org/apache/solr/cloud/ShardLeaderElectionContextBase.java

```diff
@@ -179,7 +183,16 @@ void runLeaderProcess(boolean weAreReplacement, int pauseBeforeStartMs)
           ZkStateReader.STATE_PROP, Replica.State.ACTIVE.toString());
       assert zkController != null;
       assert zkController.getOverseer() != null;
-      zkController.getOverseer().offerStateUpdate(Utils.toJSON(m));
+      DocCollection coll = zkStateReader.getCollection(this.collection);
+      if (coll == null || coll.getStateFormat() < 2 || ZkController.sendToOverseer(coll, id)) {
+        zkController.getOverseer().offerStateUpdate(Utils.toJSON(m));
+      } else {
+        PerReplicaStates prs = PerReplicaStates.fetch(coll.getZNode(), zkClient, coll.getPerReplicaStates());
+        PerReplicaStates.WriteOps writeOps = PerReplicaStates.WriteOps.flipLeader(zkStateReader.getClusterState().getCollection(collection).getSlice(shardId).getReplicaNames(), id, prs);
+        //nocommit make this debug
+        log.info("bypassed Zookeeper for leader election for {}/{}, old:{},new {} ", this.collection, shardId, currentLeader, id);
```

Comment:
This is bypassed overseer, not bypassed zookeeper, right? we're still storing that information in ZK but in a different place.

Reply:
Yes, it's true. bypassed overseer, but not zookeeper

it's a nocommit added for debugging.